frontend-api/src/test/java/uk/gov/di/authentication/frontendapi/validation/PhoneNumberCodeProcessorTest.java

```diff
@@ -82,6 +94,18 @@ void shouldReturnErrorForInvalidRegistrationPhoneNumberCode() {
                 equalTo(Optional.of(ErrorResponse.ERROR_1037)));
     }
 
+    @Test
+    void shouldNotDeleteMfaCodeFromDataStoreWhenValidRegistrationPhoneNumberCode() {
```

Comment:
```suggestion
    void shouldNotDeleteMfaCodeFromDataStoreWhenInvalidRegistrationPhoneNumberCode() {
```

This should be `whenInvalid` rather than `whenValid`

Reply:
Good spot! Updated